### ISSUE TITLE
Dirty form notifications 

### ISSUE THREAD

**Issue by tdashworth** [Capgemini/xrm-datamigration-xrmtoolbox#69]
**Is your feature request related to a problem? Please describe.**
It would be nice to receive a notification when loading a file/closing the tab if there are dirty (unsaved changes) on either of the three pages.

**Describe the solution you'd like**
Give I am a user of the plugin
And I have opened the schema/export/import page
And I make a change from the default config
When I close which page/close plugin/load a file
Then a yes/no message will open warning to lose of changes and if I want to continue

Give I am a user of the plugin
And I have opened the schema/export/import page
And I load a file
And make a change to the config
When I close which page/close plugin/load a file
Then a yes/no message will open warning to lose of changes and if I want to continue

Give I am a user of the plugin
And I have opened the schema/export/import page
And I optional load a file
And I make a change to the config
And I click save
When I close which page/close plugin/load a file
Then there will be no message and the action continues

Basically the form is not dirty when first opened, file loaded, or file saved. Any other change should make it dirty. 

Something to consider would be unsaveable fields like data type or environment. Since these would be lost when closing anyway, they shouldn't make the form dirty.
